### ISSUE TITLE
(require|disallow)SpaceAround*

### ISSUE THREAD

**Issue by TheSavior** [jscs-dev/node-jscs#1476]
Many configuration files list matching rules for `before` and `after` such as:

```
"requireSpaceBeforeBinaryOperators": ["?", "+", "/", "*", "=", "==", "===", "!=", "!==", ">", ">=", "<", "<="],
"requireSpaceAfterBinaryOperators": ["?", "+", "/", "*", "=", "==", "===", "!=", "!==", ">", ">=", "<", "<="],
```

It would be nice to have a shorthand.
```
"requireSpaceAroundBinaryOperators": ["?", "+", "/", "*", "=", "==", "===", "!=", "!==", ">", ">=", "<", "<="],
```

This rule should also respect additional operators specified for only one side.
```
"requireSpaceAroundBinaryOperators": ["?", "+", "/", "*", "=", "==", "===", "!=", "!==", ">", ">=", "<", "<="],
"requireSpaceAfterBinaryOperators": ["%"],
```

**Comment by mrjoelkemp** [comment]
Thanks for contributing the idea @TheSavior. If the goal is to avoid redundancy, you could extract the array to a variable and use it as the value for both rules within your configuration.

I do agree that `requireSpaceAroundBinaryOperators` reads well. However, if we still need to support the other two, more granular rules, then I'm not sure it's worth the added maintenance. Of course, if the community agrees, then let's do it. Any other benefits aside from avoiding config redundancy?

**Comment by TheSavior** [comment]
If you are specifying these in a .jscsrc file, how would you pull them out into a variable? My understanding is that file is just JSON.

We also already do shorthands like this. For example [disallowSpacesInFunction](http://jscs.info/rule/disallowSpacesInFunction.html) is shorthand for disallowSpacesInNamedFunctionExpression and disallowSpacesInAnonymousFunctionExpression

**Comment by mrjoelkemp** [comment]
> If you are specifying these in a .jscsrc file, how would you pull them out into a variable? My understanding is that file is just JSON.

There's the `.js` form that lets you put comments and logic in there.

> We also already do shorthands like this.

Good point.

**Comment by TheSavior** [comment]
Link to the `.js` form documentation?

**Comment by markelog** [comment]
See https://github.com/jscs-dev/node-jscs/issues/704

**Comment by hzoo** [comment]
I think we should try to have less rules - so `requireSpaceAroundBinaryOperators` could have an option for before/after/both. This would be similar to trying to combine the function spacing rules to not duplicate all the code/tests

**Comment by hzoo** [comment]
We released [3.0](https://github.com/jscs-dev/node-jscs/releases/tag/v3.0.0) recently! We also wrote a [post](https://medium.com/@markelog/jscs-end-of-the-line-bc9bf0b3fdb2#.oot3wvgpu) about how 3.0 will be our last major release (since we are deprecating the project) and that we are [merging with ESLint](http://eslint.org/blog/2016/04/welcoming-jscs-to-eslint)!

Thus we won't be working on any new features moving forward which includes new rules/options. We will continue to fix bugs and work on CST/better autofixing for current rules in the upcoming months both to help JSCS users and to gain experience that we can hopefully use in ESLint. But otherwise we both teams will be working on getting [ESLint feature parity with JSCS](https://github.com/eslint/eslint/milestones/JSCS%20Compatibility) and just a nice transition for our users.

I'm putting an `orphaned` label to note this issue isn't planned to be worked on, but might be a good contribution as a custom rule (not in JSCS core itself).

I would encourage anyone who wants the rule to try writing it themselves: I would either write a custom rule or submit an issue to ESLint and if accepted a PR. Otherwise a custom rule that won't be in core repo itself.

JSCS rules are also very similar to ESLint rules: here are some resources to get started: 

- [Project-specific Lint Rules with ESLint](https://gist.github.com/jareware/7179093)
- [Creating Custom Rules for ESLint](https://insideops.wordpress.com/2015/12/08/creating-custom-rules-for-eslint)
- [Maintaining JavaScript Code Quality with ESLint ](https://www.paypal-engineering.com/2014/12/12/maintaining-javascript-code-quality-with-eslint/)
